modules/tensor_mechanics/src/materials/FiniteStrainCrystalPlasticity.C

```diff
@@ -68,6 +80,9 @@ FiniteStrainCrystalPlasticity::FiniteStrainCrystalPlasticity(const std::string &
 
   _s0.resize(_nss);
 
+  if ( _num_slip_sys_props > 0 )
+    _slip_sys_props.resize( _nss * _num_slip_sys_props );
+
```

Comment:
Added enum to specify read type for initial values of internal variables. Depending on the option, appropriate values will be read and used. Added a test.